upload.go

```diff
@@ -127,7 +909,7 @@ func processUpload(upReq UploadRequest) (upload Upload, err error) {
 	}
 
 	if len(extension) == 0 {
-		extension = "ext"
```

Comment:
Same as above, you're basically replacing this extension = "ext" line with the mime detection.